### ISSUE TITLE
it would be cool if it works

### ISSUE THREAD

**Issue by JusttCallMeMike** [remeda/remeda#337]
<img width="586" alt="image" src="https://github.com/remeda/remeda/assets/112834462/8c4cc7d5-01d8-486d-bedf-2323be8a9837">

ts set as strictly as possible
added indexed since normal one did not worked ...and tried everything ;p

**Comment by eranhirsch** [comment]
Can you rephrase this issue? I'm not sure I understand the ask here

**Comment by TkDodo** [comment]
I think @JusttCallMeMike would want `filter(Boolean)` to filter out null and undefined values on type level. It's something that `ts-reset` can do for the native `Array.filter`. Maybe we could provide `R.filter.strict` to achieve the same ?

- https://github.com/total-typescript/ts-reset/blob/4d44ea0eee4709bcc0317592eb9ca157785122d4/src/entrypoints/filter-boolean.d.ts#L4

**Comment by eranhirsch** [comment]
Wouldn't this example simply work with `isDefined`?
```ts
const r = R.filter(children, R.isDefined);
```

**Comment by TkDodo** [comment]
yep, this works:

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wEpwGZQiOAIigFMQSATZQgbgCg6AbE+ZKKALjgEF3kBPADyoYUYADsA5nAA+ccQFdGjWXAXiKJbBMoA+OAF44AbQC6DAMYRxIuKVRL4RjADptjGCSgAKNlAA0mC7AqAAiWjoUAJR0APSxcIlwAHoA-HRAA
